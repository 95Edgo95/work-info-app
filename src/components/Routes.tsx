import WorkplacesForm from "containers/WorkplaceForm";
import WorkbookForm from "containers/WorkbookForm";
import { Switch, Route } from "react-router-dom";
import Workplaces from "containers/Workplaces";
import Workbook from "containers/Workbooks";
import withLayout from "../HOCs/withLayout";
import NotFound from "components/NotFound";
import SignIn from "containers/SignIn";
import * as React from "react";

export default function Routes(): JSX.Element {
  return (
    <Switch>
      <Route path="/sign-in" component={SignIn}/>

      <Route path="/" component={withLayout(Workbook)} exact/>
      <Route path="/workbooks" component={withLayout(Workbook)} exact/>
      <Route path="/workbooks/create" component={withLayout(WorkbookForm)} exact/>
      <Route path="/workbooks/:id/edit" component={withLayout(WorkbookForm)} exact/>

      <Route path="/workplaces/:workbookId" component={withLayout(Workplaces)} exact/>
      <Route path="/workplaces/:workbookId/create" component={withLayout(WorkplacesForm)} exact/>
      <Route path="/workplaces/:workbookId/:id/edit" component={withLayout(WorkplacesForm)} exact/>

      <Route component={withLayout(NotFound)}/>
    </Switch>
  );
}
