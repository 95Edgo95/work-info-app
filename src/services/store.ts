import createSagaMiddleware, {SagaMiddleware} from "redux-saga";
import {applyMiddleware, compose, createStore} from "redux";
import reducer from "services/reducer";
import saga from "services/saga";
import {fromJS} from "immutable";

let store: any;

const initialState: any = fromJS({});

const sagaMiddleware: SagaMiddleware<any> = createSagaMiddleware();

if (process.env && process.env.NODE_ENV !== "production") {
    const composeEnhancers: any = (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
    store = createStore(reducer, initialState, composeEnhancers(applyMiddleware(sagaMiddleware)));
} else {
    store = createStore(reducer, initialState, compose(applyMiddleware(sagaMiddleware)));
}

sagaMiddleware.run(saga);

export default store;
