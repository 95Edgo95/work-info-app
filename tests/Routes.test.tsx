import {BrowserRouter} from "react-router-dom";
import Routes from "../src/components/Routes";
import store from "../__mocks__/storeMock";
import {Provider} from "react-redux";
import { mount } from "enzyme";
import * as React from "react";

describe('Routes', () => {
  it('should render correctly', () => {
    const component = mount(
      <Provider store={store}>
        <BrowserRouter>
          <Routes />
        </BrowserRouter>
      </Provider>
    );

    expect(component).toMatchSnapshot();
    component.unmount();
  });
});
