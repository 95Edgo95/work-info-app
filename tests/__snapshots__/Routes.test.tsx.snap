// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes should render correctly 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Routes>
        <Switch>
          <Route
            component={[Function]}
            computedMatch={
              Object {
                "isExact": true,
                "params": Object {},
                "path": "/",
                "url": "/",
              }
            }
            exact={true}
            location={
              Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              }
            }
            path="/"
          >
            <Component
              history={
                Object {
                  "action": "POP",
                  "block": [Function],
                  "createHref": [Function],
                  "go": [Function],
                  "goBack": [Function],
                  "goForward": [Function],
                  "length": 1,
                  "listen": [Function],
                  "location": Object {
                    "hash": "",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  },
                  "push": [Function],
                  "replace": [Function],
                }
              }
              location={
                Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                }
              }
              match={
                Object {
                  "isExact": true,
                  "params": Object {},
                  "path": "/",
                  "url": "/",
                }
              }
            >
              <Layout>
                <div>
                  <Connect(Header)>
                    <Header
                      attemptSignOut={[Function]}
                    >
                      <div>
                        <Link
                          replace={false}
                          to="/"
                        >
                          <a
                            href="/"
                            onClick={[Function]}
                          >
                            Home
                          </a>
                        </Link>
                        <Button
                          onClick={[Function]}
                          secondary={true}
                        >
                          <button
                            className=""
                            onClick={[Function]}
                            type="button"
                          >
                            Exit
                          </button>
                        </Button>
                      </div>
                    </Header>
                  </Connect(Header)>
                  <Connect(Workbooks)
                    history={
                      Object {
                        "action": "POP",
                        "block": [Function],
                        "createHref": [Function],
                        "go": [Function],
                        "goBack": [Function],
                        "goForward": [Function],
                        "length": 1,
                        "listen": [Function],
                        "location": Object {
                          "hash": "",
                          "pathname": "/",
                          "search": "",
                          "state": undefined,
                        },
                        "push": [Function],
                        "replace": [Function],
                      }
                    }
                    location={
                      Object {
                        "hash": "",
                        "pathname": "/",
                        "search": "",
                        "state": undefined,
                      }
                    }
                    match={
                      Object {
                        "isExact": true,
                        "params": Object {},
                        "path": "/",
                        "url": "/",
                      }
                    }
                  >
                    <Workbooks
                      attemptDeleteWorkbook={[Function]}
                      attemptGetWorkbooks={[Function]}
                      history={
                        Object {
                          "action": "POP",
                          "block": [Function],
                          "createHref": [Function],
                          "go": [Function],
                          "goBack": [Function],
                          "goForward": [Function],
                          "length": 1,
                          "listen": [Function],
                          "location": Object {
                            "hash": "",
                            "pathname": "/",
                            "search": "",
                            "state": undefined,
                          },
                          "push": [Function],
                          "replace": [Function],
                        }
                      }
                      location={
                        Object {
                          "hash": "",
                          "pathname": "/",
                          "search": "",
                          "state": undefined,
                        }
                      }
                      match={
                        Object {
                          "isExact": true,
                          "params": Object {},
                          "path": "/",
                          "url": "/",
                        }
                      }
                      pagination={
                        Immutable.Map {
                          "count": 0,
                          "limit": 0,
                          "offset": 0,
                        }
                      }
                      resetWorkplaceData={[Function]}
                      workbooks={Immutable.Map {}}
                    >
                      <Filter
                        attemptGetEntities={[Function]}
                        createLink="workbooks"
                        inputs={
                          Array [
                            Object {
                              "defaultValue": "",
                              "name": "firstName",
                              "placeholder": "First Name",
                            },
                            Object {
                              "defaultValue": "",
                              "name": "lastName",
                              "placeholder": "Last Name",
                            },
                            Object {
                              "defaultValue": "",
                              "name": "passport",
                              "placeholder": "Passport",
                            },
                            Object {
                              "defaultValue": "",
                              "name": "email",
                              "placeholder": "Email",
                            },
                          ]
                        }
                      >
                        <div>
                          <Input
                            key="firstName"
                            name="firstName"
                            onChange={[Function]}
                            placeholder="First Name"
                            value=""
                          >
                            <div>
                              <div
                                className=""
                              >
                                <input
                                  name="firstName"
                                  onChange={[Function]}
                                  placeholder="First Name"
                                  type="text"
                                  value=""
                                />
                              </div>
                            </div>
                          </Input>
                          <Input
                            key="lastName"
                            name="lastName"
                            onChange={[Function]}
                            placeholder="Last Name"
                            value=""
                          >
                            <div>
                              <div
                                className=""
                              >
                                <input
                                  name="lastName"
                                  onChange={[Function]}
                                  placeholder="Last Name"
                                  type="text"
                                  value=""
                                />
                              </div>
                            </div>
                          </Input>
                          <Input
                            key="passport"
                            name="passport"
                            onChange={[Function]}
                            placeholder="Passport"
                            value=""
                          >
                            <div>
                              <div
                                className=""
                              >
                                <input
                                  name="passport"
                                  onChange={[Function]}
                                  placeholder="Passport"
                                  type="text"
                                  value=""
                                />
                              </div>
                            </div>
                          </Input>
                          <Input
                            key="email"
                            name="email"
                            onChange={[Function]}
                            placeholder="Email"
                            value=""
                          >
                            <div>
                              <div
                                className=""
                              >
                                <input
                                  name="email"
                                  onChange={[Function]}
                                  placeholder="Email"
                                  type="text"
                                  value=""
                                />
                              </div>
                            </div>
                          </Input>
                          <Button
                            onClick={[Function]}
                          >
                            <button
                              className=""
                              onClick={[Function]}
                              type="button"
                            >
                              Search
                            </button>
                          </Button>
                          <Button
                            secondary={true}
                          >
                            <button
                              className=""
                              type="button"
                            >
                              <Link
                                replace={false}
                                to="/workbooks/create"
                              >
                                <a
                                  href="/workbooks/create"
                                  onClick={[Function]}
                                >
                                  Add New
                                </a>
                              </Link>
                            </button>
                          </Button>
                        </div>
                      </Filter>
                      <CardsWrapper
                        attemptGetEntities={[Function]}
                        entities={Immutable.Map {}}
                        getDetails={[Function]}
                        getLink={[Function]}
                        getTitle={[Function]}
                        pagination={
                          Immutable.Map {
                            "count": 0,
                            "limit": 0,
                            "offset": 0,
                          }
                        }
                      >
                        <div>
                          <Component />
                        </div>
                      </CardsWrapper>
                    </Workbooks>
                  </Connect(Workbooks)>
                </div>
              </Layout>
            </Component>
          </Route>
        </Switch>
      </Routes>
    </Router>
  </BrowserRouter>
</Provider>
`;
